negotiation/negotiation.go

```diff
@@ -0,0 +1,50 @@
+package negotiation
+
+import (
+	"strconv"
+	"strings"
+)
+
+// SelectQValue selects and returns the best value from the allowed set
+// given a header with optional quality values, as you would get for an
+// Accept or Accept-Encoding header. The *first* item in allowed is preferred
```

Comment:
Do we expect people to send ordered lists of preference, or weighted lists? This code feels like it's trying to split the difference and would benefit from establishing one call pattern or the other.

Reply:
The "Q" value is an HTTP concept. See https://developer.mozilla.org/en-US/docs/Glossary/Quality_values. This code takes the user-supplied weighted list and picks from a developer-supplied ordered list to get a match for content negotiation. The developer-supplied ordered list is only ordered to break a tie should multiple user-supplied items with the same weight match. I don't think using weights for the developer side makes sense in this scenario and is more complicated than it needs to be.

Reply:
That's fair. My original comment was more that this feels like an overoptimization for a very rare use case. That said, it's not bad code, and given the potential reach of the codebase, I don't think it should be removed. This section just feels... I don't know, weirdly fiddly when compared to its surrounding complexity. I don't have any specific action items here, either, so this can be resolved if it's just "this is how it needs to operate to be compliant with the spec."